iceoryx_posh/source/popo/building_blocks/event_variable_data.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "iceoryx_posh/internal/popo/building_blocks/event_variable_data.hpp"
+
+namespace iox
+{
+namespace popo
+{
+EventVariableData::EventVariableData() noexcept
+{
+    cxx::forEach(m_activeNotifications, [](auto& id) { id.store(false, std::memory_order_relaxed); });
```

Comment:
I think I counted at least three usages of the `forEach` do you want to keep the other?

Reply:
I would keep them since I find it expressive and there are also guidelines which suggest using them. They argue like the following:
1. `for` implies some kind of order mostly from begin to end, `for_each` states that we would like to perform something for every element and order does not matter
2. `for_each` can be optimizied with an execution policy, see https://en.cppreference.com/w/cpp/algorithm/for_each this means that we state that the action is performed in multiple threads. Maybe not a use case for us at the moment but maybe later?!

I found the arguments convincing and therefore I prefer `forEach` over `for` when it makes sense.